src/libraries/System.IO.Pipes/tests/PipeStreamConformanceTests.cs

```diff
@@ -633,6 +651,26 @@ public async Task CancelTokenOn_Client_ReadWriteCancelledToken_Throws_OperationC
                 await Assert.ThrowsAnyAsync<OperationCanceledException>(() => clientWriteToken);
             }
         }
+
+        [Fact]
+        public async Task TwoServerInstances_OnceDisposed_Throws()
+        {
+            string pipeName = GetUniquePipeName();
+            var server1 = CreateServerStream(pipeName, 2);
+            using var server2 = CreateServerStream(pipeName, 2);
+
+            var wait1 = server1.WaitForConnectionAsync();
+            var wait2 = server2.WaitForConnectionAsync();
+            server1.Dispose();
+
+            using var client = CreateClientStream(pipeName);
+            await client.ConnectAsync();
+
+            await Assert.ThrowsAsync<IOException>(() => wait1);
+            await ValidateDisposedExceptionsAsync(server1);
+
+            await wait2;
```

Comment:
We should add validation that the established connection between the client and server2 is usable.

Reply:
The test currently hangs on `await wait2` when running on Unix OSes :/
I am still investigating why... but any hint is welcome.

Reply:
Doesn't that suggest the proposed fix doesn't actually fix the problem? :)

Reply:
After further analysis, it seems that this scenario is currently not possible/supported on Unix.
On `dispose` request with multiple servers, the current `accept` operation on `server1` is not (i.e. cannot be) cancelled.
https://github.com/dotnet/runtime/blob/62712ec2913a9f4a1e2e7d512a4d8b0f3299709b/src/libraries/System.IO.Pipes/src/System/IO/Pipes/NamedPipeServerStream.Unix.cs#L261-L284
So, the pipe client is connected to `server1` and this connection is now correctly closed/disposed, but it is too late. Then, the `accept` on server2 hangs forever.

I have just discovered that implementation of AcceptAsync cancellation overloads is in progress via #53340. Once available, we could add a cancellation token source to generate a token (linked to user cancellation token) for `AcceptAsync` calls that would be cancelled on `dispose` request.
Does it look reasonable to you?

Reply:
Are you saying that with unix pipes, disposing the listen socket does not cancel any outstanding accepts? Or am I misunderstanding?

Reply:
No, the ListeningSocket is shared among all server instances targeting the same pipe name. Then, it is not disposed here, as only the first server instance is disposed. So it is why I propose to cancel the accept operation, when the server instance is disposed, via a dedicated cancelation token (source).

Reply:
@stephentoub Now the PR does fix the problem ;)
Moreover, I have extended the test to validate that the established connection between the client and server2 is usable.